### PR TITLE
Linux - Envars plugin: Add function to retrieve environment variables for a specific task

### LINKED ISSUE #1411: There is no API for getting a process' environment variables on Linux
The envar plugin on Linux does all the 'work' directly in generator, which makes it impossible for other plugins to re-use:

https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/plugins/linux/envars.py#L74

We broke this out during a training so students could write + use a plugin for auto detecting maliciously set environment variables.

@gcmoreira we threw this together based on the code in generator in the plugin:

https://github.com/volatilityfoundation/volatility3/commit/2c430c9d1b94c99b71862a99deba7419ddaa7c59

Could you make this properly PR ready (typing, doc strings, etc.) and then make a PR after testing?

### PR DESCRIPTION
**Pull request by gcmoreira**
This PR addresses #1411.
It also includes some code improvements.

### CODE REVIEW

File: volatility3/framework/plugins/linux/envars.py [hunk 1]
```diff
@@ -39,76 +40,98 @@ def get_requirements(cls):
...
-            # kernel threads never have an mm as they do not have userland mappings
-            try:
-                mm = task.mm
-            except exceptions.InvalidAddressException:
-                # no mm so cannot get envars
-                vollog.debug(
-                    f"Unable to access mm for task {pid} {name} it is likely a kernel thread, will not extract any envars."
-                )
-                mm = None
+            if task.is_kernel_thread:
```
Comment (ikelos) [inline volatility3/framework/plugins/linux/envars.py:113]: Do we want to log the kernel thread like we did in the previous version?

File: volatility3/framework/plugins/linux/envars.py [hunk 2]
```diff
@@ -39,76 +40,98 @@ def get_requirements(cls):
...
+        # Read the full task environment variable buffer.
+        envar_data = proc_layer.read(env_start, env_area_size)
+
+        # Parse envar data, envars are null terminated, keys and values are separated by '='
+        envar_data = envar_data.rstrip(b"\x00")
+        for envar_pair in envar_data.split(b"\x00"):
+            try:
+                env_key, env_value = envar_pair.decode().split("=", 1)
+            except ValueError:
+                # Some legitimate programs, like 'avahi-daemon', avoid reallocating the args
```
Comment (ikelos) [inline volatility3/framework/plugins/linux/envars.py:99]: Good comment, thanks!

**Review by ikelos — APPROVED** [review]
Looks good, just wondering about silently dropping kernel tasks, but otherwise fine with it.

File: volatility3/framework/plugins/linux/envars.py (hunk 1 above)
Comment (gcmoreira) [reply volatility3/framework/plugins/linux/envars.py:113]: I don't think so. kernel threads don’t have their own memory context (mm/mm_struct), so logging that is redundant

**Pull request merged**